### PR TITLE
WMDP-194: Add basic auth-key authentication to eligibility endpoint

### PR DESCRIPTION
**Pull request by chouinar**
## Ticket
This is built ontop of the https://github.com/navapbc/wic-mt-demo-project-mock-api/pull/13 PR

https://wicmtdp.atlassian.net/browse/wmdp-194

## Changes
This adds basic authentication using a static API key which the front end eligibility screener will be able to send.

Piped through a very basic user that could be extended in a more thorough implementation.

A few minor logging and API context updates for the user.

## Context for reviewers
Note that the auth is very basic, and simply lays out a skeleton. It does map the key to a user, but it isn't connected to the DB or any other service.

## Testing
With the new environment variable, you'll need to rebuild your docker container so it gets pulled in (`make clean-volumes init start` will clear anything from old PRs out and get you setup).

Swagger will now show a place to add an auth token:
Screen Shot 2022-09-07 at 2 07 36 PM
Which if you call the API will work as it has before.

If you call without a token, you'll get the following response:
Screen Shot 2022-09-07 at 2 08 12 PM

If you call with a token that doesn't match the expected, you'll get the following response:
Screen Shot 2022-09-07 at 2 08 29 PM

If the environment variable that configures the expected token doesn't exist, you'll see:
Screen Shot 2022-09-07 at 2 15 57 PM

### CODE REVIEW

File: app/tests/api/auth/test_api_key_auth.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+def test_api_key_auth_invalid_token(api_auth_token):
+    # If you pass it the wrong token
+    with pytest.raises(
+        Unauthorized,
+        match="The server could not verify that you are authorized to access the URL requested",
+    ):
+        api_key_auth("not the right token", None)
+
+
+def test_api_key_auth_no_configuration():
```
Comment (rocketnova) [inline app/tests/api/auth/test_api_key_auth.py]: 🤔 Hmm this test is failing for me:
> AssertionError: Regex pattern 'Authentication is not setup' does not match "401 Unauthorized: The server could not verify that you are authorized to access the URL requested. You either supplied the wrong credentials (e.g. a bad password), or your browser doesn't understand how to supply the credentials required.".

**Review by rocketnova — COMMENTED** [review]
One test is failing for me. I also don't get the `Authentication is not setup properly and the user cannot be authenticated` error when I run the docker container with `API_AUTH_TOKEN` commented out.

File: app/tests/api/auth/test_api_key_auth.py (hunk 1 above)
Comment (chouinar) [reply app/tests/api/auth/test_api_key_auth.py]: Running anything inside the docker container uses your environment variables. So, inside always uses the `local.env` file. Need to make sure the env var is unset for this test to work.

Comment (rocketnova) [reply app/tests/api/auth/test_api_key_auth.py]: Yah I unset `API_AUTH_TOKEN` in `local.env` and it was still failing.

Comment (chouinar) [reply app/tests/api/auth/test_api_key_auth.py]: Did you restart the entire container before running again? Env vars only get loaded at startup

Comment (rocketnova) [reply app/tests/api/auth/test_api_key_auth.py]: I'm pretty sure I ran `make clean-volumes init start` and then `make test`. I'll try it again

Comment (rocketnova) [reply app/tests/api/auth/test_api_key_auth.py]: Hmmmmmmmm it works now 🤷

**Review by rocketnova — APPROVED** [review]
LGTM!

**Pull request merged**